### PR TITLE
Flip choices depending on the version of Symfony

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Closes #228.

### CODE REVIEW

File: Form/Type/BooleanType.php [hunk 1]
```diff
@@ -79,7 +74,21 @@ public function configureOptions(OptionsResolver $resolver)
...
                 return $options['translation_domain'];
             },
-        ));
+        );
+
+        // SF 2.7+ BC
+        if (method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
+            $choices = array_flip($choices);
+
+            // choice_as_value options is not needed of SF 3.0+
```
Comment (greg0ire) [inline Form/Type/BooleanType.php]: "needed as of" ?

File: Form/Type/EqualType.php [hunk 2]
```diff
@@ -56,13 +56,21 @@ public function configureOptions(OptionsResolver $resolver)
             self::TYPE_IS_NOT_EQUAL => $this->translator->trans('label_type_not_equals', array(), 'SonataCoreBundle'),
         );
 
-        if (!method_exists('Symfony\Component\Form\FormTypeInterface', 'setDefaultOptions')) {
+        $defaultOptions = [];
```
Comment (greg0ire) [inline Form/Type/EqualType.php]: Here is what breaks the build.

File: Form/Type/BooleanType.php [hunk 3]
```diff
@@ -73,7 +74,21 @@ public function configureOptions(OptionsResolver $resolver)
 
                 return $options['translation_domain'];
             },
-        ));
+        );
+
+        // SF 2.7+ BC
+        if (method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
+            $choices = array_flip($choices);
```
Comment (maxim-mamonov) [inline Form/Type/BooleanType.php]: Looks like this is broke BooleanFilter in the Sonata admin

Comment (Soullivaneuh) [inline Form/Type/BooleanType.php]: Could you please open an issue with more details about that?

**Pull request merged**